mbtools/generate_quiz_csv.py

```diff
@@ -0,0 +1,140 @@
+        writer.writerows(zip(*quiz_questions_csv.values()))
+
+    with open(output_path / "quiz_multichoice_answers.csv", "w") as outfile:
+        writer = csv.writer(outfile)
+        writer = csv.writer(outfile)
+        writer.writerow(quiz_multichoice_answers_csv.keys())
+        writer.writerows(zip(*quiz_multichoice_answers_csv.values()))
+
+
+def order_questions(quiz):
+    # Organize Questions by Order
+    pages = defaultdict(list)
+    for question in quiz.quiz_questions:
+        pages[question.page].append(question)
+    for page in pages.values():
+        page.sort(key=lambda h: h.slot)
+    ordered_questions = []
+    ordered_keys = list(pages.keys())
+    ordered_keys.sort()
+    for key in ordered_keys:
+        ordered_questions.extend(pages[key])
+    return ordered_questions
+
+
+def generate_quiz_data(mbz_path, output_path):
+    question_bank = utils.parse_moodle_questionbank(mbz_path)
+    quizzes = utils.parse_backup_quizzes(mbz_path)
+
+    quizzes_csv = {
+        "quiz_name": [],
+        "question_number": [],
+        "question_id": []
+    }
+    quiz_questions_csv = {
+        "id": [],
+        "text": [],
+        "type": []
+    }
+    quiz_multichoice_answers_csv = {
+        "id": [],
+        "question_id": [],
+        "text": [],
+        "grade": [],
+        "feedback": []
+    }
+
+    answer_number = 0
+    for quiz in quizzes:
+        ordered_questions = order_questions(quiz)
+
+        question_number = 0
+
+        for moodle_question in ordered_questions:
+            q_bank_question = \
+                question_bank.get_question_by_entry(
+                    question_bank_entry_id=moodle_question.qbank_entry_id,
+                    version=moodle_question.version
+                )
+
+            id_number = q_bank_question.id_number
+            text = q_bank_question.etree.xpath('./questiontext')[0].text
+            question_type = q_bank_question.question_type
+
+            quizzes_csv["quiz_name"].append(quiz.name)
+            quizzes_csv["question_number"].append(question_number)
+            quizzes_csv["question_id"].append(id_number)
+
+            if id_number not in quiz_questions_csv["id"]:
+                quiz_questions_csv["id"].append(id_number)
+                quiz_questions_csv["text"].append(text)
+                quiz_questions_csv["type"].append(question_type)
+
+                if question_type == 'multichoice':
+                    for answer in q_bank_question.multichoice_answers():
+                        answer_text = answer.answer_html_element().tostring()
+                        feedback = ""
+                        if answer.feedback_html_element() is not None:
+                            feedback = \
+                                answer.feedback_html_element().tostring()
+                        grade = answer.grade
+
+                        quiz_multichoice_answers_csv["id"].append(
+                            answer_number)
```

Comment:
Mmm, good question - I think probably it'll be cleaner to inject it as a part of enclave ETL since we do similar things there already. Just to consolidate whats happening when.

Reply:
Yep, agreed. I just updated the description of `k12-307` to incorporate adding the column there so we can remove it from this step.